src/scripts/main.js

```diff
@@ -1,3 +1,41 @@
 'use strict';
 
-// write your code here
+const logo = document.querySelector('.logo');
+
+logo.addEventListener('click', () => {
+  handleClick();
+});
+
+const handleClick = () => {
+  const promise1 = new Promise((resolve) => {
+    resolve('Promise was resolved!');
+  });
+
+  promise1.then(handleSuccess).catch(handleError);
+};
+
+const createRejectingPromise = (message) => {
+  return new Promise((resolve, reject) => {
+    setTimeout(() => {
+      reject(new Error(message));
+    }, 3000);
+  });
+};
+
+const promise2 = createRejectingPromise('Promise was rejected!');
+
+const messageDiv = document.createElement('div');
+
+document.body.appendChild(messageDiv);
+
+const handleSuccess = (message) => {
+  messageDiv.className = 'message';
+  messageDiv.textContent = message;
+};
+
+const handleError = (errorMessage) => {
+  messageDiv.className = 'message error-message';
+  messageDiv.textContent = errorMessage;
+};
+
+promise2.then(handleSuccess).catch((error) => handleError(error.message));
```

Comment:
This line is executing `promise2` immediately when the script is loaded. It's better to execute promises when needed, for example, in response to user actions. Otherwise, the promise could fail before the user has a chance to interact with the page. In this case, you might want to move this line inside the `handleClick` function.